src/Entity/Engine.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace Drupal\elastic_appsearch\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\elastic_appsearch\ElasticSearchInterface;
+use Drupal\elastic_appsearch\Utility\Database;
+
+/**
+ * Defines the Engine entity.
+ *
+ * @ConfigEntityType(
+ *   id = "engine",
+ *   label = @Translation("Engine"),
```

Comment:
`engine` is a very generic term. Shouldn't we call it `elastic_appsearch_engine`?